vaccine/states.py

```diff
@@ -254,3 +254,32 @@ async def display(self, message: Message):
                 ],
             },
         )
+
+
+class SectionedChoiceState(ChoiceState):
+    def __init__(self, *args, sections: dict, separator: Optional[str] = "", **kwargs):
+        choices = []
+        for section_choices in sections.values():
+            choices.extend(section_choices)
+
+        kwargs["choices"] = choices
+
+        super().__init__(*args, **kwargs)
+        self.sections = sections
+        self.separator = separator
+
+    @property
+    def _display_choices(self) -> str:
+        lines = []
+
+        i = 1
+        for section, choices in self.sections.items():
+            lines.append(f"*{section}*")
```

Comment:
Putting formatting here seems odd (especially since this is whatsapp specific). Surely we could put that in when we create the state in `yal/mainmenu.py`, rather than having it hardcoded here. eg. `"*CHAT SETTINGS*"` instead of `"CHAT_SETTINGS"`